core/lib/zksync_core/src/api_server/web3/backend_jsonrpsee/namespaces/eth.rs

```diff
@@ -33,8 +36,25 @@ impl EthNamespaceServer for EthNamespace {
             .map_err(|err| self.current_method().map_err(err))
     }
 
-    async fn estimate_gas(&self, req: CallRequest, block: Option<BlockNumber>) -> RpcResult<U256> {
-        self.estimate_gas_impl(req, block)
+    async fn estimate_gas(
+        &self,
+        req: CallRequest,
+        block: Option<BlockNumber>,
+        state_override: Option<StateOverride>,
+    ) -> RpcResult<U256> {
+        let validated_state_override = state_override
+            .map(|state_override| {
+                ValidatedStateOverride::try_from(state_override).map_err(|err| {
+                    ErrorObjectOwned::owned(
```

Comment:
can you make this error part of Web3Errors? https://github.com/matter-labs/zksync-era/blob/5377149384e33a11e70a7842f08d50063eb65d6f/core/lib/web3_decl/src/error.rs#L21